src/main/java/uk/ac/cam/cl/dtg/isaac/api/managers/GameManager.java

```diff
@@ -980,6 +980,52 @@ public List<GameboardItem> getNextQuestionsForFilter(final GameFilter gameFilter
     return selectionOfGameboardQuestions;
   }
 
+  /**
+   * This generates a set of five question in random.
+   *
+   * @param gameFilter - to enable search
+   * @param limit      - to provide a limit of questions
+   * @return a list of questions
+   * @throws ContentManagerException - if there is a problem accessing the content repository.
+   */
+  public List<ContentDTO> generateRandomQuestions(final GameFilter gameFilter, final int limit)
+      throws ContentManagerException {
+    List<GitContentManager.BooleanSearchClause> fieldsToMap = Lists.newArrayList();
+    fieldsToMap.add(new GitContentManager.BooleanSearchClause(
+        TYPE_FIELDNAME, uk.ac.cam.cl.dtg.segue.api.Constants.BooleanOperator.AND,
+        Collections.singletonList(QUESTION_TYPE)));
+    fieldsToMap.addAll(generateFieldToMatchForQuestionFilter(gameFilter));
+
+    List<ContentDTO> questionsToReturn = Lists.newArrayList();
+
+    while (questionsToReturn.size() < limit) {
+      var results =  this.contentManager.findByFieldNamesRandomOrder(fieldsToMap,
+          0,
+          limit,
+          null);
+
+      List<ContentDTO> generatedQuestions = results.getResults();
+
+      if (generatedQuestions.size() < limit) {
+        throw new ContentManagerException("Error in generating random questions");
+      }
+
+      for (ContentDTO question : generatedQuestions) {
+        // Only keep questions that have not been superseded or deprecated.
```

Comment:
Isn't there a way to include these two conditions in the initial search (when you get `results` for the first time)? It would make this whole thing a lot simpler: you will be able to get rid of the `while` loop as well as this `for` loop, and the code will be way more performant and readable.